jxl/src/frame/coeff_order.rs

```diff
@@ -37,6 +38,27 @@ pub const TRANSFORM_TYPE_LUT: [HfTransformType; NUM_ORDERS] = [
 
 pub const NUM_PERMUTATION_CONTEXTS: usize = 8;
 
+/// Cached natural coefficient orders for all transform types.
+/// Computed once on first access to avoid recomputation.
+static NATURAL_COEFF_ORDERS: OnceLock<[Vec<u32>; NUM_ORDERS]> = OnceLock::new();
```

Comment:
Have you tried doing a `[OnceLock<Vec<u32>>; NUM_ORDERS]` instead? that would avoid ever computing coefficient orders for large transforms that are not used.